sw/device/lib/crypto/impl/aes_gcm/ghash.c

```diff
@@ -144,6 +145,33 @@ static uint8_t reverse_bits(uint8_t byte) {
   return out;
 }
 
+uint32_t ghash_context_integrity_checksum(const ghash_context_t *ghash_ctx) {
+  uint32_t ctx;
+  crc32_init(&ctx);
+  // Compute the checksum only over a single share to avoid side-channel
+  // leakage. From a FI perspective only covering one key share is fine as
+  // (a) manipulating the second share with FI has only limited use to an
+  // adversary and (b) when manipulating the entire pointer to the key structure
+  // the checksum check fails.
+  crc32_add(&ctx, (unsigned char *)ghash_ctx->tbl0, sizeof(ghash_ctx->tbl0));
+  crc32_add(&ctx, (unsigned char *)&ghash_ctx->correction_term0,
+            sizeof(ghash_ctx->correction_term0));
+  crc32_add(&ctx, (unsigned char *)&ghash_ctx->enc_initial_counter_block0,
+            sizeof(ghash_ctx->enc_initial_counter_block0));
+  crc32_add(&ctx, (unsigned char *)&ghash_ctx->ghash_block_cnt,
+            sizeof(ghash_ctx->ghash_block_cnt));
+  return crc32_finish(&ctx);
+}
+
+hardened_bool_t ghash_context_integrity_checksum_check(
+    const ghash_context_t *ghash_ctx) {
+  if (ghash_ctx->checksum ==
+      launder32(ghash_context_integrity_checksum(ghash_ctx))) {
+    return kHardenedBoolTrue;
```

Comment:
```suggestion
    HARDENED_CHECK_EQ(ghash_ctx->checksum, ghash_context_integrity_checksum(ghash_ctx));
    return kHardenedBoolTrue;
```
would adding a redundant check make sense?